src/geojson/geojson.js

```diff
@@ -0,0 +1,494 @@
+    let tbody = table.createTBody();
+    for (let key in json) {
+        if (json.hasOwnProperty(key)) {
+
+            let row = tbody.insertRow();
+            let th = document.createElement("th");
+            let td = document.createElement("td");
+            th.appendChild(document.createTextNode(key));
+            td.appendChild(document.createTextNode(json[key]));
+            th.setAttribute("scope", "row");
+            td.setAttribute("itemprop", key);
+            row.appendChild(th);
+            row.appendChild(td);
+        }
+    }
+    return table;
+}
+
+// Takes a GeoJSON geojson and an options Object which returns a <layer-> Element
+// The options object can contain the following:
+//      label            - String, contains the layer name, if included overrides the default label mapping
+//      projection       - String, contains the projection of the layer (OSMTILE, WGS84, CBMTILE, APSTILE), defaults to OSMTILE
+//      caption          - Function | String, function accepts one argument being the feature object which produces the featurecaption string OR a string that is the name of the property that will be mapped to featurecaption
+//      properties       - Function | String | HTMLElement, a function which maps the geojson feature properties to an HTMLElement or a string that will be parsed as an HTMLElement or an HTMLElement
+//      geometryFunction - Function, A function you supply that can add classes, hyperlinks and spans to the created <map-geometry> element, default would be the plain map-geometry element
+// geojson2mapml: geojson Object <layer-> -> <layer->
+function geojson2mapml(json, options = {}, layer = null) {
+    let defaults = {
+        label: null,
+        projection: "OSMTILE",
+        caption: null,
+        properties: null,
+        geometryFunction: null
+    };
+    // assign default values for undefined options
+    options = Object.assign({}, defaults, options) ;
+
+    // If string json is received
+    if (typeof json === "string") {
+        json = JSON.parse(json);
+    }
+    let geometryType = ["POINT", "LINESTRING", "POLYGON", "MULTIPOINT", "MULTILINESTRING", "MULTIPOLYGON", "GEOMETRYCOLLECTION"];
+    let jsonType = json.type.toUpperCase();
+    let out = "";
+
+    // HTML parser
+    let parser = new DOMParser();
+
+    // initializing layer
+    if (layer === null) {
+        // creating an empty mapml layer
+        let xmlStringLayer = "<layer- label='' checked><map-meta name='projection' content='" + options.projection + "'></map-meta><map-meta name='cs' content='gcrs'></map-meta></layer->";
+        layer = parser.parseFromString(xmlStringLayer, "text/html");
+        //console.log(layer)
+        if (options.label !== null) {
+            layer.querySelector("layer-").setAttribute("label", options.label);
+        } else if (json.name) {
+            layer.querySelector("layer-").setAttribute("label", json.name);

```

Comment:
using json.name and json.title implies there are member variables that have these names, but I don't think that follows the GeoJSON spec (I haven't looked, but I guess nothing could _forbid_ it).   There are two cases: 1) "type": "Feature" and 2) "type": "FeatureCollection".  In case 1), there's often a "properties" member; in case 2) theres typically a "features" member (array).  For the purpose of this comment, we're looking to set the label attribute for the layer, and in case 1) (a single feature) it makes sense to use json.name/title OR json.properties.name/title, but for case 2) (a feature collection) it ONLY makes sense to use json.name/title

Reply:
Anyway, all that to say it's probably ok the way you've got it now.

Reply:
json.title is mentioned in the geojson spec under "[foreign member](https://www.rfc-editor.org/rfc/rfc7946#section-6.1)", so I think the need for title is there, similarly json.name is used by many GIS applications when exported as geojson, such as QGIS. 

For case 1 json.properties.name/title can be a good resource to use if available also, perhaps we can think about adding the label from properties with something similar to what we do with the "caption" option, where users can insert a property name to have the caption set to the property value.